### PR TITLE
waitQueue.compare_exchange_strong in the true loop can be weak

### PR DESCRIPTION
**Pull request by AZero13**
weak is more suitable for this while loop than strong.

### CODE REVIEW

**Comment by AtariDreams** [comment]
@DougGregor Can we please test and benchmark?

**Comment by glessard** [comment]
@swift-ci please test

File: stdlib/public/Concurrency/TaskGroup.cpp [hunk 1]
```diff
@@ -1941,13 +1941,13 @@ void TaskGroupBase::waitAll(SwiftError* bodyError, AsyncTask *waitingTask,
...
   _swift_tsan_release(static_cast<Job *>(waitingTask));
+  if (!hasSuspended) {
+    waitingTask->flagAsSuspendedOnTaskGroup(asAbstract(this));
+    hasSuspended = true;
+  }
   while (true) {
-    if (!hasSuspended) {
-      hasSuspended = true;
-      waitingTask->flagAsSuspendedOnTaskGroup(asAbstract(this));
-    }
```
Comment (kavon) [inline stdlib/public/Concurrency/TaskGroup.cpp:1948]: Looks like we can also eliminate _this_ `hasSuspended` local boolean and just call `flagAsSuspendedOnTaskGroup` before the loop, as this is the only spot it's read.

**Review by kavon — APPROVED** [review]

**Comment by kavon** [comment]
I think this is still functionally correct and may yield better performance on certain platforms. When I initially changed these two compare-exhanges to strong [in this PR](https://github.com/apple/swift/pull/40251), I was a bit overzealous.

If you could also simplify the one `hasSuspended` check then that would be great.

@ktoso for a second opinion.

File: stdlib/public/Concurrency/TaskGroup.cpp (hunk 1 above)
Comment (ktoso) [reply stdlib/public/Concurrency/TaskGroup.cpp:1948]: Yeah agreed; unconditionally setting it without the hasSuspended flat at all here seems okey to do.

**Comment by glessard** [comment]
@swift-ci please test windows platform

**Review by ktoso — APPROVED** [review]
This is okey, and we cannot remove the `hasSuspended` bool.

Thanks!

**Pull request merged**